QweixMain/Assets/Scripts/Spawn-Tile-System/PlaceByCard.cs

```diff
@@ -0,0 +1,318 @@
+/****************************************************************************
+*
+*File              : Card.cs
+* Date Created      : 2 / 16 / 23
+* Description       : This script is designed to allow the player to click on a card, drag the unit onto the board, have the board highlight appropriately, and spawn the correct object.
+* 
+*
+*  Programmer(s)     : Robert Mimms
+*  Last Modification : 3 / 16 / 23
+* Additional Notes:
+*Flow Chart URL :
+*****************************************************************************
+       (c)Copyright 2022 - 2023 by MPoweredGames - All Rights Reserved      
+****************************************************************************/
+
+
+
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlaceByCard : MonoBehaviour
+{
+    // Variables for the ghosts that follow the mouse position as the player is placing a unit
+    public GameObject following;
+    public GameObject followParent;
+    public float followSize;
+
+    // Used for determining the tile the unit will spawn at when dropped
+    public GameObject placeTile;
+    public bool placing = false;
+
+    // Represents the card being dragged onto the field and the unit that card will spawn.
+    public GameObject spawningCard;
+    public GameObject spawningUnit;
+    
+
+    // A list of the currently highlighted tiles in the grid.
+    public List<Transform> highlightedTiles = new List<Transform>();
+
+    // A reference to the GridManager script used to access the array of tiles.
+    public GameObject player;
+    public GridManager gridManager;
+    public Tile tile;
+
+    // Update is called once per frame
+    void Update()
```

Comment:
For Future reference: Functionality in Update should generally be moved to its own method and Update should be simple calling those methods. The compartmentalization makes the code easier to read and maintain.